conwayconsole/Program.cs

```diff
@@ -9,140 +9,189 @@
+                        Console.CursorVisible = false;
+                    }
+
+                    bool stop = false;
+                    DateTime lastLoopTime = DateTime.UtcNow;
+                    for (IReadableBoard board = initialBoard;
+                      !(stop || cts.IsCancellationRequested) && (options.MaxGenerations >= game.Generation);
+                      board = game.Turn(out stop))
+                    {
+                        if (!options.HideDisplay)
+                        {
+                            await Console.Out.WriteLineAsync(board.ToConsoleString(window, builder, getValueForColour));
+                            DateTime now = DateTime.UtcNow;
+                            TimeSpan elapsed = now.Subtract(lastLoopTime);
+                            TimeSpan delay = TimeSpan.FromMilliseconds(options.Delay).Subtract(elapsed);
+
+                            await Console.Out.WriteLineAsync($"{BoardConsoleExtensions.cHome}({board.Width}x{board.Height}) ({window.Width}x{window.Height}) ({window.X}, {window.Y})");
+                            if (delay > TimeSpan.Zero)
+                            {
+                                if (delay > TimeSpan.Zero)
+                                {
+                                    await Task.Delay(delay, cts.Token);
+                                }
+
+                                lastLoopTime = DateTime.UtcNow;
+                            }
+                        }
+
+                        if (save)
+                        {
+                            await Save(board);
+                            save = false;
+                            break;
+                        }
+                    }
+                }
+                catch (OperationCanceledException)
+                {
+                }
+                finally
+                {
+                    if (!options.HideDisplay)
+                    {
+                        Console.CursorVisible = true;
+                        Console.ResetColor();
+                        Console.CancelKeyPress -= HandleCancel;
+                    }
+                }
+
+                void HandleCancel(object sender, ConsoleCancelEventArgs args) => cts.Cancel();
             }
-          }
         }
-        catch (OperationCanceledException)
+
+        private static async Task<bool> YesNo(string prompt = null)
         {
+          await Console.Out.WriteAsync(prompt);
+          string response = (await Console.In.ReadLineAsync()).ToLower();
+          return response == "y" || response == "yes";
         }
-        finally
-        {       
-          if (!options.HideDisplay)
-          { 
-            Console.CursorVisible = true;
-            Console.ResetColor();
-            Console.CancelKeyPress -= HandleCancel;
-          }
-        }
+        private static async Task Save(IReadableBoard boardToSave)
+        {
+            if (boardToSave != null)
```

Comment:
The way you're executing this, boardToSave will never be null.

In any case, I don't think the `if` for that null check should be in here. It should be at the call-site (though see my other comment where I am saying that the call-site is in the wrong place)